src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationOptions.cs

```diff
@@ -152,5 +153,13 @@ public string CookieName
         /// to the client. This can be used to mitigate potential problems with very large identities.
         /// </summary>
         public IAuthenticationSessionStore SessionStore { get; set; }
+
+        CookieAuthenticationOptions IOptions<CookieAuthenticationOptions>.Value
```

Comment:
Heh, I'm not suggesting creating a base class (note that it's not abstract BTW) but a wrapper that you could use to convert a `TOptions` instance to `IOptions<TOptions>`:

    public static IApplicationBuilder UseCookieAuthentication([NotNull] this IApplicationBuilder app, CookieAuthenticationOptions options)
    {
        return app.UseMiddleware<CookieAuthenticationMiddleware>(
            new Options<CookieAuthenticationOptions>(options),
            new ConfigureOptions<CookieAuthenticationOptions>(o => { }));
    }